### PR TITLE
TaskExecutorArguments refactorization and new features.

### PR DESCRIPTION
**Pull request by anjagruenheid**
This PR changes the task executor arguments to allow users to specify global arguments (i.e., task arguments defined in the experiment configuration) as well as task-specific arguments. It also adds a new argument to retry a query until it succeeds to the task executor. Finally, the same functionality (retry until success) is enabled via max_num_retries for completeness.

### CODE REVIEW

File: src/main/java/com/microsoft/lst_bench/task/util/TaskExecutorArguments.java [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.microsoft.lst_bench.task.util;
+
+import com.microsoft.lst_bench.util.TaskExecutorArgumentsParser;
+import java.util.Map;
+import java.util.stream.Stream;
+
+public class TaskExecutorArguments {
```
Comment (netonjm) [inline src/main/java/com/microsoft/lst_bench/task/util/TaskExecutorArguments.java:22]: 

**Review by netonjm — APPROVED** [review]

**Comment by netonjm** [comment]
Tested and it seems to work. It would be great include tests or example using retry_erroneous_query_strings

**Comment by anjagruenheid** [comment]
> Tested and it seems to work. It would be great include tests or example using retry_erroneous_query_strings

The problem is how to test the retry since an error would just reoccur. Maybe mocking? Do you have any suggestions?

File: src/main/java/com/microsoft/lst_bench/task/util/TaskExecutorArguments.java [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+            .toArray(String[]::new);
+
+    this.skipExceptionStrings =
+        Stream.of(
+                this.getSkipExceptionStrings(),
+                TaskExecutorArgumentsParser.parseSkipExceptionStrings(arguments))
+            .flatMap(Stream::of)
+            .toArray(String[]::new);
+
+    Integer parsedBatchSize = TaskExecutorArgumentsParser.parseBatchSize(arguments);
```
[the remark this answers is not in the document]
Comment (anjagruenheid) [reply src/main/java/com/microsoft/lst_bench/task/util/TaskExecutorArguments.java]: Done!

File: src/main/java/com/microsoft/lst_bench/util/TaskExecutorArgumentsParser.java [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+ * limitations under the License.
+ */
+package com.microsoft.lst_bench.util;
+
+import java.util.Map;
+
+public class TaskExecutorArgumentsParser {
+
+  private static final String RETRY_ERRONEOUS_QUERY_DELIMITER = ";";
+  private static final String RETRY_ERRONEOUS_QUERY_STRINGS_KEY = "retry_erroneous_query_strings";
```
[the remark this answers is not in the document]
Comment (anjagruenheid) [reply src/main/java/com/microsoft/lst_bench/util/TaskExecutorArgumentsParser.java:23]: Done!

**Review by jcamachor — APPROVED** [review]

**Pull request merged**